drivers/sensor/lis2dux12/lis2dux12.c

```diff
@@ -0,0 +1,365 @@
+
+	lis2dux12_timestamp_set(ctx, PROPERTY_ENABLE);
+
+#ifdef CONFIG_LIS2DUX12_TRIGGER
+	if (cfg->trig_enabled) {
+		ret = lis2dux12_trigger_init(dev);
+		if (ret < 0) {
+			LOG_ERR("%s: Failed to initialize triggers", dev->name);
+			return ret;
+		}
+	}
+#endif
+
+	/* set sensor default pm and odr */
+	LOG_INF("%s: pm: %d, odr: %d", dev->name, cfg->pm, cfg->odr);
+	lis2dux12_md_t mode = {
+		.odr = cfg->odr,
+		.fs = cfg->range,
+	};
+	ret = lis2dux12_mode_set(ctx, &mode);
+	if (ret < 0) {
+		LOG_ERR("%s: odr init error (12.5 Hz)", dev->name);
+		return ret;
+	}
+
+	/* set sensor default scale (used to convert sample values) */
+	LOG_INF("%s: range is %d", dev->name, cfg->range);
+	ret = lis2dux12_set_range(dev, cfg->range);
+	if (ret < 0) {
+		LOG_ERR("%s: range init error %d", dev->name, cfg->range);
+		return ret;
+	}
+
+	return 0;
+}
+
+/*
+ * Device creation macro, shared by LIS2DUX12_DEFINE_SPI() and
+ * LIS2DUX12_DEFINE_I2C().
+ */
+
+#define LIS2DUX12_DEVICE_INIT(inst)                                                                \
+	SENSOR_DEVICE_DT_INST_DEFINE(inst, lis2dux12_init, NULL, &lis2dux12_data_##inst,           \
+				     &lis2dux12_config_##inst, POST_KERNEL,                        \
+				     CONFIG_SENSOR_INIT_PRIORITY, &lis2dux12_driver_api);
+
+/*
+ * Instantiation macros used when a device is on a SPI bus.
+ */
+
+#ifdef CONFIG_LIS2DUX12_TRIGGER
+#define LIS2DUX12_CFG_IRQ(inst)                                                                    \
+	.trig_enabled = true,                                                                      \
+	.int1_gpio = GPIO_DT_SPEC_INST_GET_OR(inst, int1_gpios, {0}),                              \
+	.int2_gpio = GPIO_DT_SPEC_INST_GET_OR(inst, int2_gpios, {0}),
+#else
+#define LIS2DUX12_CFG_IRQ(inst)
+#endif /* CONFIG_LIS2DUX12_TRIGGER */
+
+#define LIS2DUX12_SPI_OPERATION                                                                    \
+	(SPI_WORD_SET(8) | SPI_OP_MODE_MASTER | SPI_MODE_CPOL | SPI_MODE_CPHA)
+
+#define LIS2DUX12_CONFIG_SPI(inst)                                                                 \
+	{                                                                                          \
+		STMEMSC_CTX_SPI(&lis2dux12_config_##inst.stmemsc_cfg),                             \
+		.stmemsc_cfg =                                                                     \
+			{                                                                          \
+				.spi = SPI_DT_SPEC_INST_GET(inst, LIS2DUX12_SPI_OPERATION, 0),     \
+			},                                                                         \
+		.range = DT_INST_PROP(inst, range), .pm = DT_INST_PROP(inst, power_mode),          \
+		.odr = DT_INST_PROP(inst, odr),                                                    \
+		IF_ENABLED(UTIL_OR(DT_INST_NODE_HAS_PROP(inst, int1_gpio),                         \
+				   DT_INST_NODE_HAS_PROP(inst, int2_gpio)),                        \
```

Comment:
int1_gpios and int2_gpios

Reply:
Good catch, my apologies for missing this.